notify/templates.go

```diff
@@ -74,30 +73,31 @@ func (am *GrafanaAlertmanager) TestTemplate(ctx context.Context, c TestTemplates
 		}, nil
 	}
 
-	// Recreate the current template without the definition blocks that are being tested. This is so that any blocks that were removed don't get defined.
-	paths := make([]string, 0)
-	for _, name := range am.templates {
-		if name == c.Name {
-			// Skip the existing template of the same name as we're going to parse the one for testing instead.
+	// Recreate the current template replacing the definition blocks that are being tested. This is so that any blocks that were removed don't get defined.
+	var found bool
+	templateContents := make([]string, 0, len(am.templates)+1)
+	for _, td := range am.templates {
```

Comment:
nit: I think my map suggestion would simplify this code. right?

Reply:
See above [comment](https://github.com/grafana/alerting/pull/161#discussion_r1509355613)